lib/fs.js

```diff
@@ -204,18 +204,31 @@ function validateOffsetLengthWrite(offset, length, byteLength) {
   }
 }
 
-function validatePath(path, propName) {
+function validatePath(path, options) {
+  options = getOptions(options, { propName: 'path', assertFilename: false });
   let err;
 
-  if (propName === undefined) {
-    propName = 'path';
+  if (options.propName === undefined) {
+    options.propName = 'path';
   }
 
   if (typeof path !== 'string' && !isUint8Array(path)) {
-    err = new errors.TypeError('ERR_INVALID_ARG_TYPE', propName,
+    err = new errors.TypeError('ERR_INVALID_ARG_TYPE', options.propName,
                                ['string', 'Buffer', 'URL']);
   }
 
+  if (err === undefined && options.assertFilename) {
+    if (typeof path === 'string' && path[path.length - 1] === '/') {
+      err = new errors.Error('ERR_INVALID_FILE_URL_PATH',
```

Comment:
`ERR_INVALID_FILE_URL_PATH` does not look like a proper error code for this since the path here is not an URL. Can you create a new error code for it? (Also I think `getPathFromURL` can throw that error code instead of `ERR_INVALID_FILE_URL_PATH` as well). Ideally the formatter of the new error code can display the path in the message for debuggability.